frontend/src/components/MeetingRoom/ArchivingButton/ArchivingButton.tsx

```diff
@@ -18,17 +19,21 @@ export type ArchivingButtonProps = {
  * Displays a button and handles the archiving functionality. If a meeting is currently being recorded,
  * will confirm that a user wishes to stop the recording. If a meeting is not being recorded, prompts
  * the user before starting the archive.
- * @param {ArchivingButtonProps} props - The props for the component.
- *  @property {() => void} handleClick - (optional) click handler that closes the overflow menu in small view port devices.
+ * @param {ArchivingButtonProps} props - the props for the component
+ *  @property {boolean} isOverflowButton - (optional) whether the button is in the ToolbarOverflowMenu
+ *  @property {(event?: MouseEvent | TouchEvent) => void} handleClick - (optional) click handler that closes the overflow menu in small view port devices.
  * @returns {ReactElement} - The ArchivingButton component.
  */
-const ArchivingButton = ({ handleClick }: ArchivingButtonProps): ReactElement => {
+const ArchivingButton = ({
+  isOverflowButton = false,
+  handleClick,
+}: ArchivingButtonProps): ReactElement => {
   const roomName = useRoomName();
   const { archiveId } = useSessionContext();
   const isRecording = !!archiveId;
   const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
   const title = isRecording ? 'Stop recording' : 'Start recording';
-  const isSmallViewport = useIsSmallViewport();
+  const shownButtons = useShownButtons();
```

Comment:
consider renaming this hook and usage. `shownButtons` is ambiguous and really this a number, the clearer the better.